packages/playwright-test/src/index.ts

```diff
@@ -251,7 +251,11 @@ const playwrightFixtures: Fixtures<TestFixtures, WorkerFixtures> = ({
     const traceMode = normalizeTraceMode(trace);
     const defaultTraceOptions = { screenshots: true, snapshots: true, sources: true };
     const traceOptions = typeof trace === 'string' ? defaultTraceOptions : { ...defaultTraceOptions, ...trace, mode: undefined };
-    const captureTrace = shouldCaptureTrace(traceMode, testInfo);
+    // "Show browser" mode forces context reuse that lives over multiple test runs.
+    // In this case, tracesDir will be removed between runs,
+    // and ongoing tracing on the same reused context will fail with ENOENT because trace files
+    // were suddenly removed.
+    const captureTrace = shouldCaptureTrace(traceMode, testInfo) && _contextReuseMode !== 'force';
```

Comment:
I think this is misleading, we can force context reuse in the component testing and it should not affect the trace capturing, right?